start-workspace-from-url.adoc

```diff
@@ -0,0 +1,93 @@
++
+`#https://github.com/_<user_or_org>_/_<repository>_`
++
+Example: `#https://github.com/eclipse-che/che-plugin-registry`
+
+* To clone a specified branch on GitHub to the starting workspace:
++
+`#https://github.com/_<user_or_org>_/_<repository>_/tree/_<branch-name>_`
++
+Example: `#https://github.com/eclipse-che/che-plugin-registry/tree/7.39.x`
+
+* To clone the branch of a pull request on GitHub to the starting workspace:
++
+`#https://github.com/_<user_or_org>_/_<repository>_/pull/_<pull-request-id>_`
++
+Example: `#https://github.com/eclipse-che/che-plugin-registry/pull/1097`
+
+URL syntax for cloning GitLab-hosted repositories:
+
+* To clone the default branch on GitLab to the starting workspace:
++
+`#https://gitlab.com/_<user_or_org>_/_<repository>_`
+
+* To clone a specified branch on GitLab to the starting workspace:
++
+`#https://gitlab.com/_<user_or_org>_/_<repository>_/-/tree/_<branch>_`
+
+
+== Using a different editor or `IDE`
+
+The default IDE is a property of {prod-short} and default value is `eclipse/che-theia/latest`
+https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/#devfile-defaults
+
+Overriding the default editor is possible with `?che-editor=<editor-id>` parameter. The `editor-id` comes from the identifier of the IDE on the plug-in registry.
+
+Example: `https://che-server#https://github.com/example/foo?che-editor=eclipse/che-theia/next` to use the next version of Theia (build after each commit) 
+
+Some of IDEs keys are `eclipse/che-theia/latest`, `eclipse/che-theia/next`, `che-incubator/intellij-community/latest`, `che-incubator/pycharm/latest` or `che-incubator/che-code/insiders`
+
+The parameter `?che-editor` accepts a link to the devfile hosted on a public server (with Cross-Origin Resource Sharing (CORS) enabled).
+
+Example: `https://che-server#https://github.com/example/foo?che-editor=https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/insiders/devfile.yaml`
+
+== Using a different devfile file from the repository
+
+If the devfile filename is different from the default search pattern which is `.devfile.yaml` or `devfile.yaml`, it is possible to override that name.
+
+Use the parameter `?devfilePath=mydevfile.yaml` at the end of the `https://che-server#url-to-git-repository`
+
+Example: `https://che-server#https://github.com/example/foo?devfilePath=mydevfile.yaml`
+
+Short name for the parameter `devfilePath` exists: `df`
+Example: `https://che-server#https://github.com/example/foo?df=mydevfile.yaml`
+
+
+== Policy to create or reuse an existing workspace
+
+Clicks on a {prod-short} URL link create a fresh workspace when no previous matching workspace exists.
+
+Use `?new` at the end of the URL to always create a fresh workspace on every click
+
+Example: `https://che-server#https://github.com/example/foo?new` to create a new workspace from this repository.
```

Comment:
```suggestion
= URL syntax for workspace names

Until now, when you were using a URL such as `https://che#http://github.com/foo/bar`, Che would create a sequentially named workspace (example: `bar-abcde`). And each time you entered the same URL such as `https://che#http://github.com/foo/bar`, Che would create another, sequentially named workspace (example: `bar-fghij`). That has been the default behavior.

Users could also opt for a different behavior: The first time you entered a URL such as `https://che#http://github.com/foo/bar?policies.create=peruser`, Che would create a workspace named `bar`. And if you entered the same URL such as `https://che/#http://github.com/foo/bar?policies.create=peruser` again, Che would open the previously created workspace `bar` instead of creating another workspace.

Enabling the DevWorkspace Engine results in changes of such default and optional behaviors:

The new default behavior is if you enter a URL such as `https://che#http://github.com/foo/bar` and you don’t have any workspace named `bar`, Che creates a workspace named `bar` and clones the GitHub repository.

Each time you enter a URL such as `https://che#http://github.com/foo/bar` again, Che reopens the previously created workspace `bar`. (If you deleted the previously created `bar` workspace, entering `https://che#http://github.com/foo/bar` results in Che creating another workspace named `bar`.)

To create another workspace each time you enter the same URL, enter the URL appended with `?new` in the URL format `https://che#http://github.com/foo/bar?new`. Even if you already have a workspace named `bar`, each time you enter such URL, Che will create another, sequentially named workspace (example: `bar-abcde`).
```